api/v1alpha3/condition_consts.go

```diff
@@ -127,3 +124,24 @@ const (
 	// WaitingForRemediationReason is the reason used when a machine fails a health check and remediation is needed.
 	WaitingForRemediationReason = "WaitingForRemediation"
 )
+
+// Conditions and condition Reasons for the Machine's Node object
+const (
+	// MachineNodeHealthyCondition provides info about the operational state of the Kubernetes node hosted on the machine by summarizing  node conditions.
+	// If the conditions defined in a Kubernetes node (i.e., NodeReady, NodeMemoryPressure, NodeDiskPressure, NodePIDPressure, and NodeNetworkUnavailable) are in a healthy state, it will be set to True.
+	MachineNodeHealthyCondition ConditionType = "NodeHealthy"
+
+	// WaitingForNodeRefReason (Severity=Info) documents a machine.spec.providerId is not assigned yet.
+	WaitingForNodeRefReason = "WaitingForNodeRef"
```

Comment:
nit: what about 
```suggestion
	WaitingForNodeProvisioningReason = "WaitingForNodeProvisioning"
```
NodeRef is a bit too technical for a user that does not know CAPI internals

Reply:
There is already a `NodeProvisioningReason`, I think it will be confusing.
`WaitingForNodeRefReason ` is waiting for `machine.Spec.ProviderID` and
`NodeProvisioningReason` is waiting for `machine.Status.NodeRef` when there is no node found.